File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int> c;
        
        for(int i = 0; i < n; i++){
            int complement = target - nums[i];
            if(c.find(complement) != c.end()){
                return {c[complement],i};
            }
            
            c[nums[i]] = i;
        }
        return {};     
    }
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> twosum = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];
            
            if (twosum.ContainsKey(complement)) {
                return new int[] { twosum[complement], i };
            }

            if (!twosum.ContainsKey(nums[i])) {
                twosum.Add(nums[i], i);
            }
        }
        
        throw new ArgumentException("No two sum solution");
    }
}
